### ISSUE TITLE
what_if_calculator - need to clean up the errors when launching the development server

### ISSUE THREAD

**Issue by llinfeng** [UMBEE/modified-otree-snippets#1]
The following is logged upon launching a demo session for the [`what_if_calculator`](https://github.com/UMBEE/modified-otree-snippets/tree/master/what_if_calculator). To reproduce, include the `what_if_calculator/` app in `settings.py`, and start a demo session for the app.


<details>
  <summary>Click to see the logs</summary>

```
(oTree5) llinfeng 🏆 P520c ➜  modified-otree-snippets git:(master) ✗  odev 8002                                  [22/02/2| 5:22PM] Open your browser to http://localhost:8002/                                                                                        To quit the server, press Control+C.                                                                                                                                                                                                                                  Exception in ASGI application                                                                                                      Traceback (most recent call last):                                                                                                   File "/home/llinfeng/anaconda3/envs/oTree5/lib/python3.10/site-packages/anyio/streams/memory.py", line 81, in receive                return self.receive_nowait()                                                                                                     File "/home/llinfeng/anaconda3/envs/oTree5/lib/python3.10/site-packages/anyio/streams/memory.py", line 76, in receive_nowait         raise WouldBlock                                                                                                               anyio.WouldBlock                                                                                                                                                                                                                                                      During handling of the above exception, another exception occurred:                                                                                                                                                                                                   Traceback (most recent call last):                                                                                                   File "/home/llinfeng/anaconda3/envs/oTree5/lib/python3.10/site-packages/starlette/middleware/base.py", line 35, in call_next         message = await recv_stream.receive()                                                                                            File "/home/llinfeng/anaconda3/envs/oTree5/lib/python3.10/site-packages/anyio/streams/memory.py", line 101, in receive               raise EndOfStream                                                                                                              anyio.EndOfStream                                                                                                                                                                                                                                                     During handling of the above exception, another exception occurred:                                                                                                                                                                                                   Traceback (most recent call last):                                                                                                   File "/home/llinfeng/anaconda3/envs/oTree5/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 373, in run_asgi                                                                                                                                      result = await app(self.scope, self.receive, self.send)                                                                          File "/home/llinfeng/anaconda3/envs/oTree5/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__                                                                                                                                      return await self.app(scope, receive, send)                                                                                      File "/home/llinfeng/anaconda3/envs/oTree5/lib/python3.10/site-packages/starlette/applications.py", line 112, in __call__            await self.middleware_stack(scope, receive, send)                                                                                File "/home/llinfeng/anaconda3/envs/oTree5/lib/python3.10/site-packages/starlette/middleware/base.py", line 55, in __call__          response = await self.dispatch_func(request, call_next)                                                                          File "/home/llinfeng/anaconda3/envs/oTree5/lib/python3.10/site-packages/otree/middleware.py", line 23, in dispatch                   response = await call_next(request)                                                                                              
File "/home/llinfeng/anaconda3/envs/oTree5/lib/python3.10/site-packages/starlette/middleware/base.py", line 37, in call_next         
raise RuntimeError("No response returned.")                                                                                    
RuntimeError: No response returned.     
```

Note the text copied from the console needs line breaks. Here is how the logs above are shown in a console.
![image](https://user-images.githubusercontent.com/5890407/152405236-2e69d8fb-1e34-4583-9da2-a8e952edc694.png)
</details>

**Comment by llinfeng** [comment]
Related post from oTree forum: https://groups.google.com/g/otree/c/xDkukroQ2PQ/m/4UAigB4DAwAJ

**Comment by oTree-org** [comment]
Can you show the output of “pip freeze”? oTree is not even supposed to install on Python 3.10.

**Comment by llinfeng** [comment]
@oTree-org Thanks for jumping in the discussion here. I posted this issue mostly for bookkeeping purposes. 

To my surprise, without needing to change anything in that `what_if_calculator/` app, resetting the database resolved the error messages. I assume they were introduced when I was developing the demo calculator. (As I poked around and ran `otree resetdb`, well, I failed to make a copy of the local database :( I will revisit this post if I may catch the same error messages again.)

RE: using the correct version for oTree - which version should I use? Upon quickly browsing the [PyPI page](https://pypi.org/project/otree/) and the [documentation page](https://otree.readthedocs.io/en/latest/), I am afraid I cannot find the advised Python version to use. Apparently, I am not using the latest oTree release either.

----

Here is what I see upon freezing the env:
```
aiofiles==0.7.0
anyio==3.3.2
asgiref==3.4.1
certifi==2020.6.20
click==7.1.2
h11==0.12.0
idna==3.2
itsdangerous==2.0.1
MarkupSafe==2.0.1
otree==5.4.4
python-multipart==0.0.5
six==1.16.0
sniffio==1.2.0
SQLAlchemy==1.3.22
starlette==0.16.0
uvicorn==0.15.0
websockets==10.0
WTForms==2.3.3
WTForms-SQLAlchemy==0.2
```

-----


After searching for "Exception in ASGI application" on the oTree forum, I found [this post](https://groups.google.com/g/otree/c/xDkukroQ2PQ/m/4UAigB4DAwAJ) and was tempted to move the `live_method` function around. Now that resetting the database worked, I won't continue this route.
